### PR TITLE
Added Use-cases page

### LINKED ISSUE #426: ✨ Enhancement: Create the use cases page
### Is your feature request related to a problem? Please describe

This is a new feature.

### Describe the solution you'd like

As per the documentation strategy #strategy, we'd like to publish a use cases page. This page will be totally data driven, reading the information from a json file in the data folder.

I am thinking in a page similar to this one: https://nuclia.com/ai-search-use-cases/

Screenshot 2024-03-03 at 16 21 37

There will be a title and welcome message and later cards with the details of the use case.

The json storing the use cases will be based on:
* Short title
* Description supporting markdown

Please build the page with face use cases and as soon as the feature is ready I'll add the use cases.
Please build the page as per the branding and style of our page.

Similar issues: #421 #280 

### Describe alternatives you've considered

_No response_

### Additional context

_No response_

### Are you working on this?

No

### PR DESCRIPTION
**Pull request by VivekJaiswal18**
**What kind of change does this PR introduce?**
Feature added
Added use cases page and modified card component with option to support multiple text sizes.

**Issue Number:**

-  Closes #426

**Screenshots/videos:**
image

**If relevant, did you update the documentation?**
NO

**Does this PR introduce a breaking change?**

NO

### CODE REVIEW

**Comment by VivekJaiswal18** [comment]
@benjagm Please have a look
Thank you

File: components/Card.tsx [hunk 1]
```diff
@@ -7,9 +7,29 @@ interface CardProps {
...
+    medium: 'text-[1.3rem]',
+    large: 'text-[2rem]',
+  };
+  const bodyTextSizeClasses: Record<string, string> = {
+    small: 'text-[0.85rem]',
+    medium: 'text-[1rem]',
+    large: 'text-[1.5rem]',
+  };
   return (
     <div className='group relative h-full w-full max-w-lg rounded-lg border border-gray-200 bg-white p-6 px-12 shadow-3xl transition-colors delay-[150ms] ease-in-out hover:bg-slate-100'>
```
Comment (DhairyaMajmudar) [inline components/Card.tsx:34]: Here you can add the class `dark:text-black` to make the text inside cards dark-theme compatible

**Review by DhairyaMajmudar — COMMENTED** [review]
Hi Vivek you PR is looking good just left few suggestion to improve its compatibility for dark-theme and different screen sizes.

**Comment by DhairyaMajmudar** [comment]
In dark mode text inside the cards is not visible which can be solved with the help of this solution  https://github.com/json-schema-org/website/pull/589#discussion_r1536741094
image

**Comment by DhairyaMajmudar** [comment]
1. At screen size `820px` one card is breaking you can look that in below figure.

image

2. At screen size `768px` the `2xn` grid is breaking into `1xn` grid you can check that in below image.

image

**Comment by VivekJaiswal18** [comment]
Yes noticed that, and every other page using this card component is facing the same issue.

**Comment by VivekJaiswal18** [comment]
@benjagm as the card component is giving some issues regarding the responsiveness and dark mode shall I correct it in this same PR so that the issue in every page using this component would get solved.

**Comment by benjagm** [comment]
> shall I correct it in this same PR

Yes please! Thanks a lot!

**Review by benjagm — APPROVED** [review]
This is ready to merge. Thanks a lot!!

Amazing job!

**Pull request merged**